src/draw/dma2d/lv_draw_dma2d.c

```diff
@@ -0,0 +1,569 @@
+#if LV_USE_OS || LV_USE_DRAW_DMA2D_INTERRUPT
+    g_unit = draw_dma2d_unit;
+#endif
+
+#if LV_USE_OS
+    g_unit = draw_dma2d_unit;
+
+#if LV_USE_DRAW_DMA2D_INTERRUPT
+    lv_result_t res = lv_thread_sync_init(&draw_dma2d_unit->interrupt_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+
+    res = lv_thread_init(&draw_dma2d_unit->thread, LV_THREAD_PRIO_HIGH, thread_cb, 2 * 1024, draw_dma2d_unit);
+    LV_ASSERT(res == LV_RESULT_OK);
+#else
+    lv_result_t res = lv_thread_sync_init(&draw_dma2d_unit->thread_begin_polling_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+
+    res = lv_thread_init(&draw_dma2d_unit->thread, LV_THREAD_PRIO_LOW, thread_cb, 2 * 1024, draw_dma2d_unit);
+    LV_ASSERT(res == LV_RESULT_OK);
+#endif
+#endif
+
+    /* enable the DMA2D clock */
+    RCC->AHB3ENR |= RCC_AHB3ENR_DMA2DEN;
+
+    /* disable dead time */
+    DMA2D->AMTCR = 0;
+
+    /* enable the interrupt */
+    NVIC_EnableIRQ(DMA2D_IRQn);
+}
+
+void lv_draw_dma2d_deinit(void)
+{
+    /* disable the interrupt */
+    NVIC_DisableIRQ(DMA2D_IRQn);
+
+    /* disable the DMA2D clock */
+    RCC->AHB3ENR &= ~RCC_AHB3ENR_DMA2DEN;
+
+#if LV_USE_OS
+    lv_result_t res = lv_thread_delete(&g_unit->thread);
+    LV_ASSERT(res == LV_RESULT_OK);
+
+#if LV_USE_DRAW_DMA2D_INTERRUPT
+    res = lv_thread_sync_delete(&g_unit->interrupt_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+#else
+    res = lv_thread_sync_delete(&g_unit->thread_begin_polling_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+#endif
+#endif
+
+#if LV_USE_OS || LV_USE_DRAW_DMA2D_INTERRUPT
+    g_unit = NULL;
+#endif
+}
+
+#if LV_USE_DRAW_DMA2D_INTERRUPT
+void lv_draw_dma2d_transfer_complete_interrupt_handler(void)
+{
+#if LV_USE_OS
+    lv_result_t res = lv_thread_sync_signal_isr(&u->interrupt_signal);
+    LV_ASSERT(res == LV_RESULT_OK);
+#else
+    post_transfer_tasks(g_unit);
+#endif
+}
+#endif
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static int32_t evaluate_cb(lv_draw_unit_t * draw_unit, lv_draw_task_t * task)
+{
+    switch(task->type) {
+        case LV_DRAW_TASK_TYPE_FILL: {
+                lv_draw_fill_dsc_t * dsc = task->draw_dsc;
+                if(!(dsc->radius == 0
+                     && dsc->grad.dir == LV_GRAD_DIR_NONE
+                     && (dsc->base.layer->color_format == LV_COLOR_FORMAT_ARGB8888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_XRGB8888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_RGB888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_RGB565))) {
+                    return 0;
+                }
+            }
+            break;
+        default:
+            return 0;
+    }
+
+    task->preferred_draw_unit_id = DRAW_UNIT_ID_DMA2D;
+    task->preference_score = 0;
+
+    return 0;
+}
+
+static int32_t dispatch_cb(lv_draw_unit_t * draw_unit, lv_layer_t * layer)
+{
+    lv_draw_dma2d_unit_t * draw_dma2d_unit = (lv_draw_dma2d_unit_t *) draw_unit;
```

Comment:
We should check if the `draw_unit` is busy. See [here](https://github.com/lvgl/lvgl/blob/d7b82c79668d7b31d52f6535360eff15396eade5/src/draw/renesas/dave2d/lv_draw_dave2d.c#L344)